src/engine/sim_engine/engine.rs

```diff
@@ -40,7 +47,50 @@ impl Engine for SimEngine {
         let redundancy = calculate_redundancy!(redundancy);
 
         match self.pools.get_by_name(name) {
-            Some(_) => Ok(CreateAction::Identity),
+            Some((_, pool)) => {
+                let input_devices = blockdev_paths
+                    .iter()
+                    .cloned()
+                    .collect::<HashSet<&Path, RandomState>>();
+
+                let blockdev_paths: Vec<_> = pool
```

Comment:
If you can fuse construction of the currently used devices set into a single expression, that would be great.

Reply:
This was a change made due to lifetimes. We have two options here. We can either keep it as is, or we can switch from `Path`s to `PathBuf`s as they are owned. I'll try with `PathBuf`s and see if I can get it into one expression. I also disliked having one variable just for the sake of lifetimes so I'll see what I can do.

Reply:
The switch to `PathBuf` was successful.

Reply:
Resolved.